sdk/src/main/java/com/mapbox/maps/MapController.kt

```diff
@@ -217,12 +218,17 @@ internal class MapController : MapPluginProviderDelegate, MapControllable {
   }
 
   override fun addWidget(widget: Widget) {
+    val weakThis = WeakReference(this)
+    widget.setRepaintTrigger {
+      weakThis.get()?.renderer?.scheduleRepaint()
```

Comment:
do we need `scheduleRepaint` again at L226? since we are doing `scheduleRepaint()` when adding widget here?

Also maybe change `weakThis` to `mapControllerWeakReference` to understand better.

Reply:
> do we need scheduleRepaint again at L226?

yes, we should do repaint when widget is added to the map. The `setRepaintTrigger` is only useful when user modify the widget itself, but not when the constructed widget is added to the map.

> Also maybe change weakThis to mapControllerWeakReference to understand better.

It feels weakThis is meaningful enough as we are creating the WeakReference with `this`.